infra/feast-operator/api/v1alpha1/featurestore_types.go

```diff
@@ -503,10 +503,11 @@ type ServerConfigs struct {
 }
 
 // RegistryServerConfigs creates a registry server for the feast service, with specified container configurations.
+// +kubebuilder:validation:XValidation:rule="self.restAPI == true || self.grpc == true || !has(self.grpc)", message="At least one of restAPI or grpc must be true"
```

Comment:
in this situation it shouldn't be a check that at least 1 is true, because we set GRPC by default for the user... so the user _could_ leave them both blank if they wanted.

here, we **must** to check for a scenario where the user explicitly sets both to `false`.

Reply:
something like this _should_ work -
```go
// +kubebuilder:validation:XValidation:rule="!(self.restAPI == false && self.grpc == false)", message="restAPI and grpc cannot both be false"
```